tests/notifications/test-class-yoast-notification.php

```diff
@@ -0,0 +1,322 @@
+				'capability_check' => 'any',
+			)
+		);
+
+		$this->assertFalse( $subject->display_for_current_user() );
+	}
+
+	/**
+	 * Test any without matches.
+	 */
+	public function test_match_all_pass() {
+
+		$me = wp_get_current_user();
+		$me->add_cap( 'bla' );
+		$me->add_cap( 'foo' );
+
+		$subject = new Yoast_Notification(
+			'message',
+			array(
+				'id'               => 'id',
+				'capabilities'     => array(
+					'bla',
+					'foo',
+				),
+				'capability_check' => 'all',
+			)
+		);
+
+		$this->assertTrue( $subject->display_for_current_user() );
+
+		$me->remove_cap( 'bla' );
+		$me->remove_cap( 'foo' );
+	}
+
+	/**
+	 * Test any without matches.
+	 */
+	public function test_match_all_fail() {
+
+		$subject = new Yoast_Notification(
+			'message',
+			array(
+				'id'               => 'id',
+				'capabilities'     => array(
+					'bla',
+					'foo',
+				),
+				'capability_check' => 'all',
+			)
+		);
+
+		$this->assertFalse( $subject->display_for_current_user() );
+	}
+
+	/**
+	 * Test capability filter
+	 */
+	public function test_wpseo_notification_capabilities() {
+		$filter_function = array( $this, 'add_wpseo_notification_capabilities' );
+
+		$me = wp_get_current_user();
+
+		$capabilities = $this->add_wpseo_notification_capabilities();
+		foreach ( $capabilities as $capability ) {
+			$me->add_cap( $capability );
+		}
+
+		$me->remove_cap( 'manage_options' );
+
+		$subject = new Yoast_Notification( 'message', array( 'id' => 'id', 'capabilities' => 'manage_options' ) );
+
+		$this->assertFalse( $subject->display_for_current_user() );
+
+		add_filter( 'wpseo_notification_capabilities', $filter_function );
+		$this->assertTrue( $subject->display_for_current_user() );
+		remove_filter( 'wpseo_notification_capabilities', $filter_function );
+	}
+
+	/**
+	 * Test capability filter
+	 */
+	public function test_wpseo_notification_capability_match() {
+		$filter_function = array( $this, 'add_wpseo_notification_capabilities' );
+		$match_function  = array( $this, 'add_wpseo_notification_capability_check' );
+
+		// Only add 1 of the provided capabilities. We need 'any' to match this.
+		$capabilities = $this->add_wpseo_notification_capabilities();
+
+		$me = wp_get_current_user();
+		$me->add_cap( $capabilities[0] );
+		$me->remove_cap( $capabilities[1] );
+
+		$me->remove_cap( 'manage_options' );
+
+		$subject = new Yoast_Notification( 'message', array( 'id' => 'id', 'capabilities' => array( 'manage_options' ) ) );
+
+		$this->assertFalse( $subject->display_for_current_user() );
+
+		add_filter( 'wpseo_notification_capability_check', $match_function );
+		add_filter( 'wpseo_notification_capabilities', $filter_function );
+		$this->assertTrue( $subject->display_for_current_user() );
+		remove_filter( 'wpseo_notification_capabilities', $filter_function );
+		remove_filter( 'wpseo_notification_capability_check', $match_function );
+	}
+
+	/**
+	 * Invalid filter return value
+	 */
+	public function test_invalid_filter_return_values() {
```

Comment:
You can actually test this without using the filter and just putting `capabilities => 1` in the constructor.